include/numerics/petsc_solver_exception.h

```diff
@@ -130,13 +139,29 @@ PETSC_BEGIN_END(VecGhostUpdate) // VecGhostUpdateBeginEnd
   } while (0)
 
 #define LibmeshPetscCall2(comm, ...)                                                               \
+  do                                                                                               \
+  {                                                                                                \
+    PetscErrorCode libmesh_petsc_call_ierr;                                                        \
+    libmesh_petsc_call_ierr = __VA_ARGS__;                                                         \
+    LIBMESH_CHKERR2(comm.get(), libmesh_petsc_call_ierr);                                          \
+  } while (0)
+
+#define LibmeshPetscCallA(comm, ...)                                                               \
   do                                                                                               \
   {                                                                                                \
     PetscErrorCode libmesh_petsc_call_ierr;                                                        \
     libmesh_petsc_call_ierr = __VA_ARGS__;                                                         \
     LIBMESH_CHKERR2(comm, libmesh_petsc_call_ierr);                                                \
   } while (0)
 
+#define LibmeshPetscCallQ(...)                                                                     \
```

Comment:
With all these macros now, please add comments about when developers should use which